src/onegov/feriennet/app.py

```diff
@@ -95,9 +95,14 @@ def banner(self, request, id):
         """ Randomly returns the html to one of the available booking banners.
 
         """
+        language = request.locale[:2]
         candidates = [
             sponsor for sponsor in self.sponsors
-            if getattr(sponsor, 'banners', None) and id in sponsor.banners
+            if (
+                getattr(sponsor, 'banners', None)
+                and id in sponsor.banners
+                and sponsor.banners['bookings']['src'][language]
```

Comment:
Does ['bookings']['src''] exist in every case or could it raise an AttributeError? A more error-prone possibility would be to use sponsor.banners.get('bookings', {}).get('src', {}).get('language', None).